solr/core/src/java/org/apache/solr/api/ContainerPluginsRegistry.java

```diff
@@ -422,7 +421,10 @@ public static Class getConfigClass(ConfigurablePlugin<?> o) {
       for (Type type : interfaces) {
         if (type instanceof ParameterizedType) {
           ParameterizedType parameterizedType = (ParameterizedType) type;
-          if (parameterizedType.getRawType() == ConfigurablePlugin.class) {
+          Type rawType = parameterizedType.getRawType();
+          if (rawType == ConfigurablePlugin.class ||
```

Comment:
Is it worth keeping `rawType == ConfigurablePlugin.class` in the if condition? It's covered by the second operand of the OR.

Reply:
Right, I wrestled with this expressions to make it work and forgot to simplify.